### PR TITLE
Added a11y Storybook addon

### PR DESCRIPTION
**Pull request by nickytonline**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [ ] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update
- [x] Developer  experience
- [x] Accessbility

## Description

We already have testing of a11y violations in our frontend tests via [jest-axe](), but this is a great addon regardless. You can see pretty descriptive information about a11y violations/passes etc. as you build your component out.

Documentation about our usage of Storybook in general will be in another PR at some point as we're still adding things to Storybook. We were previously being held back for some addons due to being on an older version of Preact. That is no longer the case.

## QA Instructions, Screenshots, Recordings

image

To test:

* Navigate to a Storybook story in the Netlify preview site for Storybook, e.g. the [Article from an organization story](https://deploy-preview-8814--storybookdevto.netlify.app/?path=/story/4-app-components-article-standard--with-organization), to see the a11y panel
* Click on the Accessibility panel to view a11y violations/passes

## Added tests?

- [ ] yes
- [x] no, because they aren't needed
- [ ] no, because I need help

## Added to documentation?

- [ ] docs.dev.to
- [ ] readme
- [x] no documentation needed

## [optional] What gif best describes this PR or how it makes you feel?

Kermit and Animal from the Muppets hugging

### CODE REVIEW

File: app/javascript/.storybook/preview.js [hunk 1]
```diff
@@ -50,6 +51,7 @@ const themeSwitcherDecorator = (storyFn) => (
 );
 
 addDecorator(themeSwitcherDecorator);
+addDecorator(withA11y);
```
Comment (nickytonline) [inline app/javascript/.storybook/preview.js:54]: Enables the a11y panel within Storybook for all Storybook stories.

File: app/javascript/.storybook/main.js [hunk 2]
```diff
@@ -6,6 +6,7 @@ module.exports = {
     '@storybook/addon-knobs',
     '@storybook/addon-actions',
     '@storybook/addon-links',
+    '@storybook/addon-a11y/register',
```
Comment (nickytonline) [inline app/javascript/.storybook/main.js:9]: Registers the Storybook addon for a11y.

File: yarn.lock [hunk 3]
```diff
@@ -12266,6 +12304,14 @@ redoc@2.0.0-rc.29:
     tslib "^1.11.1"
     url-template "^2.0.8"
 
+redux@^4.0.1:
```
Comment (Ridhwana) [inline yarn.lock:12307]: Is it weird that it added redux, is that a dependancy of the addon?

Comment (nickytonline) [reply yarn.lock:12307]: Redux is a dependency of Storybook. There must have been an update for some reason.

**Pull request merged**